iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp

```diff
@@ -280,6 +291,23 @@ class LLVMAOTTargetBackend final : public TargetBackend {
       objectFiles.push_back(std::move(objectFile));
     }
 
+    if (!options_.staticLibraryOutput.empty()) {
+      if (objectFiles.size() != 1) {
+        // Static library output only supports single object libraries.
+        return targetOp.emitError() << "Static library generation unsupported "
```

Comment:
(or better wording would be "generating static libraries from multiple object files is not supported") - though really this can't happen at all as we only generate a single object file so this could be an assert instead :)